controller/appcontroller.go

```diff
@@ -1020,7 +1066,24 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 			},
 		},
 	)
-	return informer, lister
+	err := informer.AddIndexers(cache.Indexers{
+		orphanedIndex: func(obj interface{}) (i []string, e error) {
+			app, ok := obj.(*appv1.Application)
+			if !ok {
+				return nil, nil
+			}
+
+			proj, err := ctrl.getAppProj(app)
+			if err != nil {
+				return nil, nil
```

Comment:
did you want to be returning an error here and on line 1073?

Reply:
I did initially but in this case controller panics. If this happens controller adds error condition to the app and don't reconcile it.